### ISSUE TITLE
resizewindow dispatcher is invalid?

### ISSUE THREAD

**Issue by stratosgear** [dawsers/hyprscroller#99]
I am trying a:

```
➜  hyprctl dispatch scroller:resizewindow onehalf
Invalid dispatcher
```

How is this supposed to work?

I am trying to set some keybinds for setting specific window widths?

Running on nixos unstable on hyprscroller  0-unstable-2024-11-29

**Comment by cpiber** [comment]
hyprscroller does not have such a dispatcher. In the readme, Hyprland's `resizewindow` dispatcher is referenced, which has been renamed to `resizewindowpixel`

**Comment by dawsers** [comment]
Yes, the README is wrong, the dispatcher is `resizeactive` which is  then referenced further down in the README in the [Key Bindings](https://github.com/dawsers/hyprscroller#key-bindings) example.

The way *hyprscroller* works is not by setting the width with a keystroke. You usually define a set of standard widths you want to use, a default one, and then cycle through them using `scroller:cyclewidth next/prev`. For example, in your config:

```
plugin {
  scroller {
    column_default_width = onehalf
    column_widths = twothirds one onethird onehalf
  }
}

bind = $mainMod, equal, scroller:cyclewidth, next
bind = $mainMod, minus, scroller:cyclewidth, prev
bind = $mainMod SHIFT, equal, scroller:cycleheight, next
bind = $mainMod SHIFT, minus, scroller:cycleheight, prev
```

That would set a default width of `onehalf`, and cycling widths of `twothirds one onethird onehalf` each time you press `mainMod +/-`.

**Comment by stratosgear** [comment]
OK makes sense, but still, why are you not offering the ability to go from 5/8s to 1/8s (as an example), instantaneously with a keypress, but are forcing the users to cycle between the available widths...?  

It seems that you already have all the important required pieces in place, especially all the calculations of onethird, fiveeights etc for each monitor, that I cannot do (as easily) with plain hyprland.

So basically what I ask is for a direct `resizeto` (or however you might want to call it) dispatcher that accepts one of the predefined column widths.  In certain cases I know exactly the width that I want, and keeping it within the predefined ones, I kinda guarantee that the remaining columns will continue fitting in my screen in a reasonable manner.

What a shame to miss just that one dispatcher.. :)  I'm I missing a potential implementation difficulty?

Thanks btw for a very-very nice plugin!

**Comment by dawsers** [comment]
I didn't implement that because Hyprland already provides the feature, as explained above. I don't use it, mainly because I would need a lot of key bindings to cover all those combinations, and also good memory to remember which one means what. I'd rather use +,- to find the size I need, even if I have to press them more than once (most times once or twice).

So if you want to use the workflow you mention, assign key bindings to Hyprland's own dispatcher which you can use within hyprscroller with no problems. For example (adapt to your monitor's resolution):

```
bind = $mainMod, 1, resizeactive, exact 700 1666
bind = $mainMod, 2, resizeactive, exact 1000 1666
bind = $mainMod, 3, resizeactive, exact 1300 1666
bind = $mainMod, 4, resizeactive, exact 1600 1666
...
```

And if what you get is not exactly what you want, you can always use `fitsize` to make any set of windows fit the width of the monitor exactly, re-scaling them automatically as needed.

**Comment by stratosgear** [comment]
Apologies for being such a bother, I don't want to bother you any more than required...

Why not offer the dispatcher, even if it not useful to you? (maybe the answer is in this same question, I do not know... :smile:  )

It seems that you have already done all the hard work.  Especially the hard work of splitting the widths of the monitor in parts of 8s and 6s.  And doing the additionally, even nicer, hard work of extending it to multiple monitors.

In my usecase:

```
hyprctl monitors | grep -a1 Monitor
Monitor DP-1 (ID 0):
	1920x1080@60.00000 at 2560x0
--
Monitor HDMI-A-1 (ID 1):
	2560x1080@74.99100 at 0x0
```

the use of the keybinds as you suggested, does not work nicely, because of different sized monitors, that prevent the columns to nicely align.... :(

You already offer 20+ dispatchers, is there room for one more?

Again, the plugin is EXCELLENT as it is.  Maybe I'm looking for the EXCELLENT+?

I'm not going to bother you any more, in any case.  Happy holidays!  :santa:
